internal/plugin/factory.go

```diff
@@ -16,6 +16,11 @@ import (
 	"github.com/hashicorp/waypoint-plugin-sdk/internal-shared/pluginclient"
 )
 
+// Indicates that the process is running inside an ondemand runner. This information
+// is passed to the plugins so they can configure themselves specially for this
+// unique context.
+var InsideODR bool
```

Comment:
Ah hah. I wonderrrrr if we want to introduce a `internal/odr` package or something that exports an `InODR() bool` that we set globally and just use that. Kind of like an `os.GOARCH` feeling to it. That way if we have other ODR-specific logic we can just depend on that environmental-feeling package rather than sending this around everywhere.

Not crtical to this PR, but wanted to note it.

Reply:
Sure, we can that later. The other way to handle it is to refactor how we call Factory to have factory level configuration. I almost did that but the PR was getting too long.